geode-core/src/main/java/org/apache/geode/distributed/internal/membership/adapter/ServiceConfig.java

```diff
@@ -12,161 +12,325 @@
  * or implied. See the License for the specific language governing permissions and limitations under
  * the License.
  */
-package org.apache.geode.distributed.internal.membership.gms;
+package org.apache.geode.distributed.internal.membership.adapter;
 
 
+import java.net.InetAddress;
+
 import org.apache.geode.distributed.Locator;
 import org.apache.geode.distributed.internal.DistributionConfig;
 import org.apache.geode.distributed.internal.InternalLocator;
+import org.apache.geode.distributed.internal.membership.gms.api.MembershipConfig;
 import org.apache.geode.internal.admin.remote.RemoteTransportConfig;
 import org.apache.geode.internal.net.SocketCreator;
 
-public class ServiceConfig {
-
-  /** stall time to wait for concurrent join/leave/remove requests to be received */
-  public static final long MEMBER_REQUEST_COLLECTION_INTERVAL =
-      Long.getLong(DistributionConfig.GEMFIRE_PREFIX + "member-request-collection-interval", 300);
-
-  /** in a small cluster we might want to involve all members in operations */
-  public static final int SMALL_CLUSTER_SIZE = 9;
+public class ServiceConfig implements MembershipConfig {
 
   /** various settings from Geode configuration */
   private final long joinTimeout;
   private final int[] membershipPortRange;
   private final long memberTimeout;
 
-  private boolean isReconnecting;
-  private Integer lossThreshold;
+  private final boolean isReconnecting;
+  private final Integer lossThreshold;
   private final Integer memberWeight;
-  private boolean networkPartitionDetectionEnabled;
+  private final boolean networkPartitionDetectionEnabled;
   private final int locatorWaitTime;
 
   /** the configuration for the distributed system */
   private final DistributionConfig dconfig;
 
   /** the transport config from the distribution manager */
   private final RemoteTransportConfig transport;
+  private final boolean locatorsPreferredAsCoordinators;
+
+  public ServiceConfig(RemoteTransportConfig transport, DistributionConfig theConfig) {
+    this.dconfig = theConfig;
+    this.transport = transport;
+    this.isReconnecting = (transport.getOldDSMembershipInfo() != null);
+
+    long defaultJoinTimeout = 24000;
+    if (isReconnecting || theConfig.getLocators().length() > 0 && !Locator.hasLocator()) {
```

Comment:
The adapter package will stay in core. So I guess technically this will remove the need to fix the static lookup if we just want geode-membership to not do any static lookups. But maybe we still want to fix this.